### PR TITLE
chore: Updating API providers with new types

### PR DESCRIPTION
**Pull request by davidterpay**

### CODE REVIEW

File: pkg/math/math.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ func Min[V int | int64 | uint64 | int32 | uint32](vals ...V) V {
 }
 
 // Float64StringToBigInt converts a float64 string to a big.Int.
-func Float64StringToBigInt(s string, decimals int) (*big.Int, error) {
+func Float64StringToBigInt(s string, decimals uint64) (*big.Int, error) {
```
Comment (davidterpay) [inline pkg/math/math.go:23]: Decimals are all now uint64

File: providers/apis/binance/api_handler.go [hunk 2]
```diff
@@ -10,115 +10,124 @@ import (
...
+		_, resolvedOk := resolved[ticker]
+		_, unresolvedOk := unresolved[ticker]
+
+		if !resolvedOk && !unresolvedOk {
+			unresolved[ticker] = fmt.Errorf("no response")
+		}
 	}
 
-	return providertypes.NewGetResponse[slinkytypes.CurrencyPair, *big.Int](resolved, unresolved)
+	return providertypes.NewGetResponse(resolved, unresolved)
```
Comment (aljo242) [inline providers/apis/binance/api_handler.go]: Should probably `NewGetPricesResponse`

File: providers/apis/coinbase/api_handler.go [hunk 3]
```diff
@@ -8,103 +8,115 @@ import (
...
-	market, ok := h.cfg.Market.CurrencyPairToMarketConfigs[cp.String()]
+	ticker := tickers[0]
+	market, ok := h.market.TickerConfigs[ticker.String()]
 	if !ok {
-		return "", fmt.Errorf("unknown currency pair %s", cp)
+		return "", fmt.Errorf("unknown ticker %s", ticker.String())
 	}
 
-	return fmt.Sprintf(h.cfg.API.URL, market.Ticker), nil
+	return fmt.Sprintf(h.api.URL, market.OffChainTicker), nil
```
Comment (aljo242) [inline providers/apis/coinbase/api_handler.go:77]: Are we always ensuring that `h.api.URL` will be a valid format string?

i.e. a string containing "%s" or something in it?

Comment (davidterpay) [reply providers/apis/coinbase/api_handler.go:77]: yea we do

File: providers/apis/binance/api_handler.go (hunk 2 above)
Comment (davidterpay) [reply providers/apis/binance/api_handler.go]: :sunglasses:

File: providers/apis/coinbase/api_handler.go (hunk 3 above)
Comment (aljo242) [reply providers/apis/coinbase/api_handler.go:77]: we are so smart

**Comment by aljo242** [comment]
`make format` :(

**Review by aljo242 — APPROVED** [review]
lgtm

just need a `make format`

**Pull request merged**